### PR TITLE
chore(java_extractor): refactor java extractor for readability

### PR DESCRIPTION
**Pull request by shahms**
Refactor runJavaAnalysisToExtractCompilationDetails to make data flow and dependencies more clear by extracting behavior into named functions and classes.

### CODE REVIEW

File: kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java [hunk 1]
```diff
@@ -703,110 +712,39 @@ private AnalysisResults runJavaAnalysisToExtractCompilationDetails(
       throws ExtractionException {
 
     AnalysisResults results = new AnalysisResults();
+    DiagnosticCollector<JavaFileObject> diagnosticCollector = new DiagnosticCollector<>();
+    CompilationUnitCollector compilationCollector = new CompilationUnitCollector();
 
-    // We will initialize and run the Javac compiler to detect which dependencies
-    // the current compilation has.
-    JavaCompiler compiler = findJavaCompiler();
-    if (compiler == null) {
```
Comment (schroederc) [inline kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java]: Why was this removed?

Comment (shahms) [reply kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java]: It was not; it was moved into `getJavaCompiler`
